hugr/src/types/type_param.rs

```diff
@@ -258,9 +275,17 @@ impl TypeArg {
 
```

Comment:
Ok, I agree checking in validation is a good move. There are three options:
* A new SignatureError variant. I.e. a "public" error. This is the "nicest", but if this occurs, this is a coding bug on our part rather than a user / Hugr construction error, right? I mean, even deserialization should use `new_var_use` etc. rather than directly constructing `TypeArgVariable`. Hence....
* An `assert`. This is what I've done, but it may be the worst of three options; if this occurs in "production", user will not only use Eq-totalness, but indeed, the rest of their program run too ;-).
* `debug_assert`

I went for the second but I'm moving towards the third, unless you strongly push me towards the first...

Reply:
Re `check_typevar_decl`, we can't do it there as the *correct* representation as a `TypeArg::Type` of a `Type(Enum)::(Row)Variable` eventually comes back to `check_typevar_decl` with the correct/expected form of TypeParam(::List). So I think that's in the right place - variables bound by TypeParams can be just Types not TypeArgs.